File: 백준/Silver/1181. 단어 정렬/단어 정렬.java
import java.util.*;
import java.io.*;

public class Main {

	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int N = Integer.parseInt(br.readLine());
		
		HashSet<String> set = new HashSet<>();
		for(int i = 0; i < N; i++) {
			set.add(br.readLine());
		}
		
		List<String> list = new ArrayList<>(set);
		
		list.sort(Comparator
				.comparingInt(String::length)
				.thenComparing(Comparator.naturalOrder()));
		
		StringBuilder sb = new StringBuilder();
		for(String s: list){
			sb.append(s).append('\n');
		}
		
		System.out.println(sb);
	}

}

File: 백준/Silver/1181. 단어 정렬/단어 정렬.kt
package org.example

fun main() {
    val br = System.`in`.bufferedReader()
    val sb = StringBuilder()
    val n = br.readLine().toInt()
    // 중복 단어 제거 필요
    val hashSet = HashSet<String>()
    repeat(n) {
        hashSet.add(br.readLine())
    }
    hashSet.sortedWith(compareBy({ it.length }, { it })).forEach {
        sb.append("$it\n")
    }
    println(sb)
    br.close()
}